include/osrm/errorcodes.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef ERRORCODES_HPP
+#define ERRORCODES_HPP
+
+#include <string>
+
+namespace osrm
+{
+
+/**
+ * Various error codes that can be returned by OSRM internal functions.
+ * Note: often, these translate into return codes from `int main()` functions.
+ *       Thus, do not change the order - if adding new codes, append them to the
+ *       end (but leave _ENDMARKER_ in the last position), so the code values do
+ *       not change for users that are checking for certain values.
+ */
+enum ErrorCode
```

Comment:
Hmm not sure why you need to `static_cast<int>` the enum below this should be backed by `int` by default

[`Unscoped Enums/1`](http://en.cppreference.com/w/cpp/language/enum)